polars/polars-core/src/frame/groupby/aggregations/dispatch.rs

```diff
@@ -205,4 +205,39 @@ impl Series {
         };
         self.restore_logical(out)
     }
+
+    #[doc(hidden)]
+    // Takes the single element (if there is one) per group.
+    // Panics if there are multiple elements.
+    pub unsafe fn _agg_single(&self, groups: &GroupsProxy) -> PolarsResult<Series> {
```

Comment:
We don't need this. We can check the length of `GroupsProxy` before we aggregate. It should be equal to `value_col`, if not we can throw an error.

If the lengths are as expected we can continue with an `agg_first`.

Reply:
Sure - is there no risk of having a group with no elements?

There seems to be logic to handle this, but I can't find any tests which hit it

https://github.com/pola-rs/polars/blob/8a03fbe81bcc4f427412e8b720c9b9767f26f2a4/polars/polars-core/src/frame/groupby/aggregations/dispatch.rs#L53-L54

Reply:
This is not possible from groups coming from a `groupby` as every group has at least the element that initiated that group.

This can only happen in aggregation where users apply a filter in the expressions, but that's not the case here.

Reply:
That's clear, thanks for explaining!

I'll see if I can increase coverage in a separate PR then